docs/custom-workload.md

```diff
@@ -0,0 +1,115 @@
+# Deploying a custom workload
+
+## Example workload
+
+A typical windows deployment would look like this.
+
+```shell script
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: win-webserver
+  name: win-webserver
+spec:
+  selector:
+    matchLabels:
+      app: win-webserver
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: win-webserver
+      name: win-webserver
+    spec:
+      tolerations:
+      - key: "os"
+        value: "Windows"
+        Effect: "NoSchedule"
+      containers:
+      - name: windowswebserver
+        image: mcr.microsoft.com/windows/servercore:ltsc2019
+        imagePullPolicy: IfNotPresent
+        command:
+        - powershell.exe
+        - -command
+        - $listener = New-Object System.Net.HttpListener; $listener.Prefixes.Add('http://*:80/'); $listener.Start();Write-Host('Listening at http://*:80/'); while ($listener.IsListening) { $context = $listener.GetContext(); $response = $context.Response; $content='<html><body><H1>Red Hat OpenShift + Windows Container Workloads</H1></body></html>'; $buffer = [System.Text.Encoding]::UTF8.GetBytes($content); $response.ContentLength64 = $buffer.Length; $response.OutputStream.Write($buffer, 0, $buffer.Length); $response.Close(); };
+        securityContext:
+          runAsNonRoot: false
+          windowsOptions:
+            runAsUserName: "ContainerAdministrator"
+      nodeSelector:
+        kubernetes.io/os: windows
+```
+
+For more information about windows workloads, visit
+[the docs](<https://docs.openshift.com/container-platform/4.12/windows_containers/scheduling-windows-workloads.html>).
+
+## Configure the base image
+
+It is very important to set the correct base image version for your test workload.
+Each version of the WMCO uses a specific windows operating system version, and each workload has its own base image.
+
+To know which windows version your release uses, check the defaultSKU in
+[azure.go](<https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/providers/azure/azure.go#L21>)
+for the WMCO release you’re testing with.
+To know which base image your release uses, check the
+[getWindowsServerContainerImage()](<https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/network_test.go#L460>)
+function in network_test.go for the WMCO release you’re testing with.
```

Comment:
This should be generic and not Azure or e2e specific. It boils down what Windows Server version has been used in the MachineSet / BYOH image.